src/row.rs

```diff
@@ -188,7 +188,8 @@ impl Row {
     pub(crate) fn draw(&self, offset: usize, max_len: usize, buffer: &mut String) {
         let mut current_hl_type = HLType::Normal;
         let chars = self.render.chars().skip(offset).take(max_len);
-        for (c, (i, mut hl_type)) in chars.zip(self.hl.iter().enumerate().skip(offset)) {
+        let mut cx = self.render.chars().take(offset).map(|c| c.len_utf8()).sum();
```

Comment:
You should be able to use `rx2cx` to compute `cx`: `let cx = self.rx2cx[i]` (you can rename `i` to `rx` if you want).

Reply:
Unformately, in this case, `i` is **not** `rx`.  For example, 
```
use unicode_width::UnicodeWidthChar;

fn main() {
    let s = "中文";
    s.chars().for_each(|c| println!("{}, {}", c, c.width().unwrap_or(1)));
}
```
We got
```
中, 2
文, 2
```
So for `文`, its `rx` supposes to be `2` instead of `1`.